config/src/config/network_config.rs

```diff
@@ -215,18 +218,40 @@ impl NetworkConfig {
         }
     }
 
+    #[allow(deprecated)]
     pub fn discovery_methods(&self) -> Vec<&DiscoveryMethod> {
-        // TODO: This is a backwards compatibility feature.  Deprecate discovery_method
-        if self.discovery_method != DiscoveryMethod::None && !self.discovery_methods.is_empty() {
-            panic!("Can't specify discovery_method and discovery_methods")
-        } else if self.discovery_method != DiscoveryMethod::None {
-            vec![&self.discovery_method]
-        } else {
-            self.discovery_methods
-                .iter()
-                .filter(|method| &&DiscoveryMethod::None != method)
-                .collect()
+        if self.discovery_method.is_some() {
+            warn!("discovery_method is deprecated, use discovery_methods instead")
+        }
+
+        // Check for invalid combinations of discovery methods.
+        let using_none_with_others = self
```

Comment:
When we remove this function, we should eventually add this check to the config sanitizer 😄